API/Sensor/business.py

```diff
@@ -42,43 +50,213 @@ def get_list_of_product(self, request):
+
+        first = ean_list[0]
+        product = products_list[[i for i, _ in enumerate(products_list) if _.__dict__['id'] == first['id']][0]]
+        add = ''
+        if product.status == 2:
+            add = ' Le produit peut ne plus se trouver à cet endroit.'
+        return HttpResponse(200).custom({'state': f'Nous avons trouvé: {product.product_name}, dans: {product.position}.{add}'})
+
+    # lie un sensor à un user uniquement s'il n'existe déjà pas
+    def link_sensor_user(self, request, id_sensor, id_user):
+        if not request:
+            abort(400)
+        elif id_sensor <= 0 or id_user <= 0:
+            return HttpResponse(403).error(ErrorCode.UNK)
+
+        sensor = session.query(Sensor).filter(Sensor.id == id_sensor).first()
+        if sensor:
+            return HttpResponse(403).error(ErrorCode.SENSOR_EXISTS)
+
+        new_sensor = Sensor(
+            id=id_sensor,
+            date_insert=datetime.datetime.now(),
+            date_update=datetime.datetime.now(),
+            is_active=True,
+            id_user=id_user,
+            type=False,
+            name=False,
+            sensorcol=False
+        )
+        try:
+            session.begin()
+            session.add(new_sensor)
+            session.commit()
+        except Exception as e:
+            session.rollback()
+            session.flush()
+            return HttpResponse(500).error(ErrorCode.DB_ERROR, e)
+        return HttpResponse(201).success(SuccessCode.SENSOR_LINKED, {})
+
+    # permet de (re)nommer un sensor s'il est déjà lié à un utilisateur
+    def name_sensor(self, request, id_sensor, name):
+        if not request:
+            abort(400)
+        elif id_sensor <= 0 or len(name) == 0:
+            return HttpResponse(403).error(ErrorCode.UNK)
+
+        sensor = session.query(Sensor).filter(Sensor.id == id_sensor).first()
+        if not sensor:
+            return HttpResponse(403).error(ErrorCode.SENSOR_NFIND)
+
+        try:
+            session.begin()
+            session.query(Sensor).filter(Sensor.id == id_sensor).update({'name': name})
+            session.commit()
+        except Exception as e:
+            session.rollback()
+            session.flush()
+            return HttpResponse(500).error(ErrorCode.DB_ERROR, e)
+        return HttpResponse(201).success(SuccessCode.SENSOR_NAME_UPDATED, {})
+
+    # renvoie la liste de tous les capteurs
+    def get_list_name(self, request):
+        if not request:
+            abort(400)
+        sensor_list = session.query(Sensor).filter(Sensor.id_user == self.user_connected, Sensor.is_active).all()
+        #sensor_list = session.query(Sensor).all()
+        if not sensor_list or len(sensor_list) == 0:
+            return HttpResponse(500).error(ErrorCode.SENSOR_NDETECTED)
+
+        res = 'Voici la liste des endroits où je peux trouver des produits :'
+        for x in sensor_list:
+            sensor = to_dict(x)
+            if len(res) > 61:
```

Comment:
61 correspond à la longueur initiale

Reply:
Donc ?

Reply:
Il faut soit mettre un commentaire dans le code pour expliquer, soit récuperer la longueur initial avant le for